storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -1767,6 +1900,169 @@
                                   topology-id
                                   (doto (GetInfoOptions.) (.set_num_err_choice NumErrorsChoice/ALL))))
 
+      ;;Blobstore implementation code
+      (^String beginCreateBlob [this
+                                ^String blob-key
+                                ^SettableBlobMeta blob-meta]
+        (let [session-id (uuid)]
+          (.put (:blob-uploaders nimbus)
+            session-id
+            (->> (ReqContext/context)
+              (.subject)
+              (.createBlob (:blob-store nimbus) blob-key blob-meta)))
```

Comment:
Here and elsewhere, use helper function `get-subject` instead of code like `(->> (ReqContext/context) (.subject))`